src/main/java/com/enonic/lib/guillotine/Synchronizer.java

```diff
@@ -0,0 +1,11 @@
+package com.enonic.lib.guillotine;
+
+import com.enonic.xp.script.ScriptValue;
+
+public class Synchronizer
```

Comment:
IMO synchronisation is broken. 
Invalidation happens without this class involved.
Either guard all modifications of map by it or don't synchronise at all

Reply:
Thank you. I've added sync for invalidate